var/spack/repos/builtin/packages/mpich/package.py

```diff
@@ -100,6 +100,11 @@ def url_for_version(self, version):
 '-Wl,-flat_namespace'.'''
     )
 
+    variant('cuda', default=False, description="Enable CUDA support")
+    depends_on('cuda', when='+cuda')
+    conflicts('+cuda', when='device=ch3')
+    conflicts('+cuda', when='@:3.4')
```

Comment:
Prefer a dependent variant to a conflict:
```suggestion
    variant('cuda', default=False, when='@3.5:', description="Enable CUDA support")
    depends_on('cuda', when='+cuda')
    conflicts('+cuda', when='device=ch3')
```

Reply:
oops, didn't realize you were reviewing this. i'll drop it

Reply:
No worries, I just happened to be browsing. Happy to leave it to you since I typically have worked with OpenMPI...